### PR TITLE
Added chapter 3

### PR DESCRIPTION
**Pull request by caposcar1998**
Exercise for chpater 3

### CODE REVIEW

File: chapter3/src/order.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+		Id:          id,
+		orderLines:  orderLine,
+		total:       total,
+		dateCreated: dateCreated,
+		user:        user,
+	}, nil
+
+}
+
+func AppendOrderLine(orderLine OrderLine, order *Order) {
```
Comment (LE-Blythe) [inline chapter3/src/order.go:51]: You can make functions part of the object like this: 

`func (order *Order) AppendOrderLine(orderLine OrderLine)`

Then you can call this function like this: 

`order.AppendOrderLine(orderLine)`

Comment (caposcar1998) [reply chapter3/src/order.go:51]: Ohh that cool, thnx!

**Review by stefanosKarantin — COMMENTED** [review]
Very good work mate! Keep it up! 🦾 
It is not important for the sake of this exercise but I would prefer to have the struct definition and methods implementations of `orders` and `orderlines` in different respective packages. This way you would have a clear seperation of the domain entities.

File: chapter3/src/order.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+	user        string
+}
+
+type OrderLine struct {
+	Id          string
+	item        string
+	dateCreated time.Time
+	quantity    int16
+	unitPrice   float32
+}
```
Comment (fpapadopou) [inline chapter3/src/order.go:23]: Typically you don't define types in main package. This package is only meant for the code that runs your program. You could move these types into a dedicated package, which becomes more important as your codebase grows.

Comment (fpapadopou) [reply chapter3/src/order.go:23]: Assuming you do introduce a new package for orders, the `CreateOrder` method would become a public `New` method in that package.

**Review by fpapadopou — COMMENTED** [review]
Looks good, left a comment about packaging.

File: chapter3/src/order.go (hunk 2 above)
Comment (caposcar1998) [reply chapter3/src/order.go:23]: Ohh thats true, thnx!

**Pull request merged**